telegram/src/main/kotlin/com/github/kotlintelegrambot/network/retrofit/converters/ChatIdConverterFactory.kt

```diff
@@ -0,0 +1,22 @@
+package com.github.kotlintelegrambot.network.retrofit.converters
+
+import com.github.kotlintelegrambot.entities.ChatId
+import com.github.kotlintelegrambot.entities.ChatId.Id
+import com.github.kotlintelegrambot.entities.ChatId.Username
+import java.lang.reflect.Type
+import retrofit2.Converter
+import retrofit2.Retrofit
+
+class ChatIdConverterFactory : Converter.Factory() {
+    override fun stringConverter(type: Type, annotations: Array<Annotation>, retrofit: Retrofit): Converter<ChatId, String>? {
```

Comment:
Have you checked that the API operation works with both id types? I'm not 100% sure the Telegram Bot API will accept the numeric id as a string

Reply:
Yes, I checked this. 
We use form urlencoded body and everything should be converted to string here (hence we override `stringConverter` method only).
If we decide to send json body where string and int representation are different, then we will need to use another deserialization approach.

Reply:
Oh, got it. I guess we'll also need a gson serializer/deserializer for it at some point then.